scalardb-test/src/main/java/kelpie/scalardb/transfer/TransferProcessor.java

```diff
@@ -61,23 +80,49 @@ public void close() {
           builder.add(txId, Json.createArrayBuilder().add(ids.get(0)).add(ids.get(1)).build());
         });
 
-    setState(Json.createObjectBuilder().add("unknown_transaction", builder.build()).build());
+    setState(
+        Json.createObjectBuilder()
+            .add("num_updates", numUpdates.get())
+            .add("unknown_transaction", builder.build())
+            .build());
   }
 
-  private void transfer(DistributedTransaction transaction, int fromId, int toId, int amount)
+  private void transfer(
+      DistributedTransaction transaction,
+      int fromId,
+      int fromType,
+      int toId,
+      int toType,
+      int amount)
       throws TransactionException {
-    int fromType = 0;
-    int toType = 0;
-    if (fromId == toId) {
-      toType = 1; // transfer between the same account
+    int fromBalance = 0;
+    if (isWriteSkewTest.get()) {
```

Comment:
I feel it would be more maintainable to create a workload specific processor since it makes it very complicated if it handles a new workload, for example, for read-only anomalies.
As a downside of that, there could introduce some redundant code, which must be the reasons you chose the current way, so generalization of TransactionProcessor might be needed to properly do that.

What do you think ?

Reply:
Thank you! I separated the tests to 2 processors and 2 checkers.